### PR TITLE
增加对网易云音乐.uc缓存格式和QQ音乐.cache缓存格式的支持

### PR DESCRIPTION
**Pull request by qq1010903229**
增加对网易云音乐.uc缓存格式和QQ音乐.cache缓存格式的支持

### CODE REVIEW

**Comment by qq1010903229** [comment]
[网易云缓存uc文件样本.zip](https://github.com/unlock-music/unlock-music/files/6573660/uc.zip)
压缩包密码34

**Comment by qq1010903229** [comment]
[QQ音乐缓存cache文件样本.zip](https://github.com/unlock-music/unlock-music/files/6573666/QQ.cache.zip)

**Comment by emmmx** [comment]
It seems doesn't work on QQ music cache files for encrypted files.
You may try to call QMCDecrypt to handle such cases
[sample(.mgg.cache).zip](https://github.com/unlock-music/unlock-music/files/6581020/sample.mgg.cache.zip)

File: src/decrypt/qmccache.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        const buffer = new Uint8Array(await GetArrayBuffer(file));
+        let length = buffer.length
+        for (let i = 0; i < length; i++) {
+             buffer[i] ^= 0xf4
+             if(buffer[i] <= 0x3f)buffer[i] = buffer[i] * 4;
+             else if(buffer[i] <= 0x7f)buffer[i] = (buffer[i] - 0x40) * 4 + 1;
+             else if(buffer[i] <= 0xbf)buffer[i] = (buffer[i] - 0x80) * 4 + 2;
+             else buffer[i] = (buffer[i] - 0xc0) * 4 + 3;
+        }
+        ext = SniffAudioExt(buffer, raw_ext);
```
Comment (emmmx) [inline src/decrypt/qmccache.ts:20]: Need to handle cases like .mgg.cache

File: src/decrypt/common.ts [hunk 2]
```diff
@@ -16,6 +18,9 @@ export async function CommonDecrypt(file: FileInfo): Promise<DecryptResult> {
         case "ncm":// Netease Mp3/Flac
             rt_data = await NcmDecrypt(file.raw, raw_filename, raw_ext);
             break;
+        case "uc":// Netease Cache
+            rt_data = await NcmCacheDecrypt(file.raw, raw_filename, "mp3");
```
Comment (emmmx) [inline src/decrypt/common.ts:22]: Do not use "mp3" as raw_ext, it caused the conversion to fail.

```golang
rt_data = await NcmCacheDecrypt(file.raw, raw_filename, raw_ext);
```

**Review by emmmx — CHANGES_REQUESTED** [review]
Great job, but fix the following problem please.

**Comment by emmmx** [comment]
Thanks for your work.
Right now it's still not working. I will merge to a non-main branch and fix.
Since your commit is too fragmented, I would squash and merge

**Pull request merged**